### ISSUE TITLE
using insert_many and on_conflict ignore, panics on empty unwrap

### ISSUE THREAD

**Issue by Samiisd** [SeaQL/sea-orm#1193]
## Description

When using `insert_many` with `OnConflict` directive and all of the entries have conflicts, I get the following error:

`thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', /home/perso/.cargo/registry/src/github.com-1ecc6299db9ec823/sea-orm-0.10.1/src/executor/insert.rs:119:54`

source code: https://github.com/SeaQL/sea-orm/blob/master/src/executor/insert.rs#L119

It seems to fail whenever all entries being inserted are conflicting, hence ignored, i.e. no entries are added.
However, if at least one of the entries is not conflicting, no errors.

## Steps to Reproduce

1. create a db with unique constraint on a field name e.g.
2. run a query, using sea-orm, insert_many + on_conflict do_nothing, twice

Example query:
```rust
let models = vec![symbol::Model {name: Set("hello"), ..Default::default()}, symbol::Model {name: Set("bye"), ..Default::default()};
symbol::Entity::insert_many(models)
                    .on_conflict(
                        OnConflict::column(symbol::Column::Name)
                            .do_nothing()
                            .to_owned(),
                    )
                    .exec(&state.db)
                    .await
```

### Expected Behavior

No error.

### Actual Behavior

panic on unwrap on None.

### Reproduces How Often

This is not random, can always be reproduced.

## Versions

```
│   │   ├── sea-orm v0.10.1
│   │   │   ├── sea-orm-macros v0.10.1 (proc-macro)
│   │   │   ├── sea-query v0.27.1
│   │   │   │   ├── sea-query-derive v0.2.0 (proc-macro)
│   │   │   ├── sea-query-binder v0.2.1
│   │   │   │   ├── sea-query v0.27.1 (*)
│   │   │   ├── sea-strum v0.23.0
│   │   │   │   └── sea-strum_macros v0.23.0 (proc-macro)
│   │   ├── sea-query v0.27.1 (*)
│   │   └── sea-orm-migration v0.10.1
│   │       ├── sea-orm v0.10.1 (*)
│   │       ├── sea-orm-cli v0.10.1
│   │       │   ├── sea-schema v0.10.2
│   │       │   │   ├── sea-query v0.27.1 (*)
│   │       │   │   └── sea-schema-derive v0.1.0 (proc-macro)
│   │       ├── sea-schema v0.10.2 (*)
│   ├── sea-orm v0.10.1 (*)
```

os: Ubuntu 22
db: postgresql

**Comment by tyt2y3** [comment]
I feel like there is a number of duplicate to this issue.
I suspect it's how we handle returning.
Again, `panic on unwrap on None.` looks bad.

**Comment by sh-soltanpour** [comment]
Hi,
I'm thinking about a possible way to handle this in code. 
The case is that the query is executed successfully but due to the `on_conflict` policy, nothing is inserted in the database. 

So we can change the type of `last_insert_id` to `Option` in [`InsertResult`](https://github.com/SeaQL/sea-orm/blob/master/src/executor/insert.rs#L26), so it can be `None` in cases like this. The downside is this would be a breaking change for those who are directly getting `last_insert_id` from `InsertResult`. 

The other backward-compatible way that comes to my mind is using some magic values for `last_insert_id` like `-1`, which is not a good way I believe.

Would this be a good way to handle this? @tyt2y3 @billy1624

**Comment by tyt2y3** [comment]
I think it is already being resolved for Postgres in https://github.com/SeaQL/sea-orm/pull/1021
@billy1624 can we extend the logic and test case to others as well?

**Comment by sh-soltanpour** [comment]
> I think it is already being resolved for Postgres in #1021 @billy1624 can we extend the logic and test case to others as well?

Oh thanks for mentioning it, I've just realized that PR. Reading the code for that PR, a question came to my mind. When we have an insert query with a `on conflict do nothing` constraint, sometimes the query may not insert anything but it is executed successfully anyway, so is it reasonable to return `Error` instead of a `Result`?

**Comment by tyt2y3** [comment]
This should have been addressed in 0.11 with https://github.com/SeaQL/sea-orm/pull/1403

**Comment by tyt2y3** [comment]
@billy1624 I think it is worthy to mention somewhere in the docs, where the error variant also has to be checked.
